### PR TITLE
feat(stepper): setting up stepper component with basic behavior

### PR DESCRIPTION
**Pull request by soykje**
**TASK**: https://jira.ets.mpi-internal.com/browse/SPA-48

### Description, Motivation and Context
Initialize component with basic behavior.

### Types of changes
- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 🧾 Documentation
- [x] 🧪 Test
- [x] 💄 Styles

image

### CODE REVIEW

File: packages/components/stepper/src/Stepper.doc.mdx [hunk 1]
```diff
@@ -20,12 +21,113 @@ npm install @spark-ui/stepper
...
+import { Stepper } from '@spark-ui/stepper'
 ```
 
 ## Props
 
-<ArgTypes of={Stepper} />
+<ArgTypes
+  of={Stepper}
+  description="A container for all the stepper component parts."
+  exclude={[
```
Comment (Powerplex) [inline packages/components/stepper/src/Stepper.doc.mdx:32]: I'm curious why we need to manually exclude this, are they not excluded by our global storybook filter by default ? (We didn't have to do this on other components)

File: packages/components/stepper/src/useStepper.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}: UseStepperArgs): UseStepperReturn => {
+  const state = useNumberFieldState({
+    ...rest,
+    isDisabled: rest.disabled,
+    isReadOnly: rest.readOnly,
+    isRequired: rest.required,
+    locale,
+  })
+
+  const { groupProps, inputProps, incrementButtonProps, decrementButtonProps } = useNumberField(
```
Comment (Powerplex) [inline packages/components/stepper/src/useStepper.ts:19]: Personal preference, but on other components we expose the whole api returned by the library (wether it is Zag, PopperJS, etc), it avoids destructuring, simplifies typing (eliminated the need to use `Pick<>` and allows us to use/explore the full API in each part of the compound.

WDYT ?

File: packages/components/stepper/src/Stepper.tsx [hunk 3]
```diff
@@ -1,7 +1,144 @@
...
+      decrementAriaLabel,
+      ...stepperProps
+    },
+    forwardedRef
+  ) => {
+    const inputRef = useRef(null)
+
+    const {
+      groupProps,
+      inputProps: _inputProps,
```
Comment (Powerplex) [inline packages/components/stepper/src/Stepper.tsx:40]: Why the `_` prefix ? Why not also on `groupProps` ?

File: packages/components/stepper/src/Stepper.tsx [hunk 4]
```diff
@@ -1,7 +1,144 @@
...
+        return childrenArray
+          .filter(isValidElement)
+          .find(child =>
+            (child.type as FC & { displayName?: string }).displayName?.includes(elementDisplayName)
+          )
+      },
+      [children]
+    )
+
+    const incrementBtnFromChildren = findElement('Stepper.IncrementButton')
```
Comment (Powerplex) [inline packages/components/stepper/src/Stepper.tsx:66]: The `findElement` pattern is a anti-pattern that goes against our composability rules, we decided to stop using it. Is there a reason you must use it ?

File: packages/components/stepper/src/useStepper.ts (hunk 2 above)
Comment (soykje) [reply packages/components/stepper/src/useStepper.ts:19]: Passing a huge and obscure pack of props could also be less readable I think. I guess it is a matter of personal preference but using (and exposing) only the necesary parts also allows us to keep our code more easy to read and understand.

**Review by Powerplex — APPROVED** [review]

File: packages/components/stepper/src/Stepper.doc.mdx (hunk 1 above)
Comment (Powerplex) [reply packages/components/stepper/src/Stepper.doc.mdx:32]: I agree with you, but I thought native react props were automatically omitted with our setup

File: packages/components/stepper/src/Stepper.tsx (hunk 4 above)
Comment (Powerplex) [reply packages/components/stepper/src/Stepper.tsx]: Let's discuss it when you come back :) I agree it is necessary in some cases (I think only Select and Snackbar it is justified). I think it should be eliminated from Input too, but it is a breaking change and is not an emergency.

**Pull request merged**